ffi/diplomat/src/tinystr.rs

```diff
@@ -88,6 +88,13 @@ macro_rules! impl_tinystr_convert {
                 unsafe { TinyAsciiStr::from_bytes_unchecked(other.data.to_le_bytes()) }
             }
         }
+        impl From<&$ffi_t> for TinyAsciiStr<$n> {
+            fn from(other: &$ffi_t) -> Self {
+                // Safety: The data field is private, and the only way to construct
```

Comment:
issue: no it's not; it's public over FFI

We have no concept of private wrappers in Diplomat right now. They're either opaque types (costly), or simple structs. E.g. in JS you can just pass `{id: 1}` to functions.

We could add private wrapper structs but I would rather we migrate to HIR first.